torch/_inductor/codegen/comm_buffer_planning.py

```diff
@@ -0,0 +1,248 @@
+# mypy: allow-untyped-defs
+from __future__ import annotations
+
+import collections
+import dataclasses
+from typing import Any, List, Optional, TYPE_CHECKING
+
+import torch
+
+from ..virtualized import V
+from .memory_planning import Allocation, AllocationPool, LiveRange, TemporalSplit
+from .wrapper import (
+    CommBufferAllocateLine,
+    CommBufferFreeLine,
+    CommBufferLine,
+    NullLine,
+)
+
+
+if TYPE_CHECKING:
+    from .. import ir
+    from ..utils import IndentedBuffer
+
+
+# NOTE [comm buffer planning]
+#
+# This file contains the memory planning logic for comm buffers. Compared to
+# regular buffer planning, Inductor leverages the allocator's "persistent
+# allocation" capability to meet the stringent requirements for registered
+# buffer communication (see NOTE [lowering-time collective optimization] for
+# details regarding the requirements).
+#
+# Comm buffer planning is a collaborative process between Inductor and the
+# allocator. Inductor is responsible for planning comm buffers within a graph.
+# For each (comm_buffer_type, group_name) pair, Inductor uses a single,
```

Comment:
can you define `(comm_buffer_type, group_name) pair` (or explain what the differerent types are for)?  I assume group_name refers to process_group?  (also, why do different PGs need to have different allocations? bc we have to tie the registration to a particular ncclcomm?